src/tests/conditions.test.js

```diff
@@ -56,3 +51,308 @@ it('Should allow check of a nested property in a conditional', () => {
+          hours: {
+            type: 'number',
+            description: 'Any value works.',
+          },
+        },
+        allOf: [
+          {
+            if: {
+              properties: { is_full_time: { const: 'yes' } },
+              required: ['is_full_time'],
+            },
+            then: {
+              properties: {
+                hours: {
+                  description: 'We recommend 8 hours.',
+                },
+              },
+            },
+          },
+        ],
+      },
+      { strictInputType: false }
+    );
+
+    // By default the attribute is set the base value.
+    expect(fields[1].description).toBe('Any value works.');
+
+    // Given "Yes" it applies the conditional attribute
+    expect(handleValidation({ is_full_time: 'yes', hours: 4 }).formErrors).toBeUndefined();
+    expect(fields[1].description).toBe('We recommend 8 hours.');
+
+    // Changing to "No", it applies the base value again.
+    expect(handleValidation({ is_full_time: 'no', hours: 4 }).formErrors).toBeUndefined();
+    expect(fields[1].description).toBe('Any value works.');
+
+    // Changing back to "Yes", it sets the attribute again
+    expect(handleValidation({ is_full_time: 'yes', hours: 8 }).formErrors).toBeUndefined();
+    expect(fields[1].description).toBe('We recommend 8 hours.');
+  });
+
+  it('Update a nested attribute', () => {
+    const { fields, handleValidation } = createHeadlessForm(
+      {
+        properties: {
+          is_full_time: { type: 'string', oneOf: [{ const: 'yes' }, { const: 'no' }] },
+          hours: {
+            type: 'number',
+            presentation: {
+              inputType: 'number',
+              anything: 'info',
+            },
+          },
+        },
+        allOf: [
+          {
+            if: {
+              properties: { is_full_time: { const: 'yes' } },
+              required: ['is_full_time'],
+            },
+            then: {
+              properties: {
+                hours: {
+                  presentation: {
+                    anything: 'danger',
+                  },
+                },
+              },
+            },
+          },
+        ],
+      },
+      { strictInputType: false }
+    );
+
+    // By default the attribute is set the base value.
+    expect(fields[1].anything).toBe('info');
+
+    // Given "Yes" it applies the conditional attribute
+    expect(handleValidation({ is_full_time: 'yes' }).formErrors).toBeUndefined();
+    expect(fields[1].anything).toBe('danger');
+
+    // Changing to "No", it applies the base value again.
+    expect(handleValidation({ is_full_time: 'no' }).formErrors).toBeUndefined();
+    expect(fields[1].anything).toBe('info');
+
+    // Changing back to "Yes", it sets the attribute again
+    expect(handleValidation({ is_full_time: 'yes' }).formErrors).toBeUndefined();
+    expect(fields[1].anything).toBe('danger');
+  });
+
+  it('Keeps existing attributes in matches that dont change the attr', () => {
```

Comment:
bug: Our internal Storybook tests spot this scenario as failing. I'm looking into a fix right now.

Reply:
I think I cracked it, ready for review!

Change summary: We cannot compare the `conditionalAttrs` to the current`field` alone, we also need to compare it to the original field (the ones in the root).